### PR TITLE
Test utils

### PR DESCRIPTION
**Pull request by mmscibor**
Added tests to `utils.py`. I have omitted the `log_message`, which should likely be rethought, as well as `load_json_index` - which I'm not sure if getting `CWD` is the correct approach. 

This should cover most of the tests though.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                   expected_missing,
+                   expected_inv_checksum):
+    index_path = os.path.join("tests/indexes", test_index)
+    with open(index_path) as index_file:
+        test_index = json.load(index_file)
+
+    missing_files, invalid_checksums = utils.validator(test_index, "tests/resources/")
+
+    assert expected_missing == len(missing_files)
+    assert expected_inv_checksum == len(invalid_checksums)
```
Comment (drubinstein) [inline tests/test_utils.py]: Any reason why you can't compare against the expected missing files and invalid checksums and not just the length?

Comment (mmscibor) [reply tests/test_utils.py]: Good point, will make this more thorough

File: tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+from mirdata import utils
+
+import json
+import pytest
+
+if sys.version_info.major == 3:
+    builtin_module_name = 'builtins'
+else:
+    builtin_module_name = '__builtin__'
```
Comment (mmscibor) [inline tests/test_utils.py]: @drubinstein maybe a better way to do this?

Comment (drubinstein) [reply tests/test_utils.py]: `from future import builtins`

**Comment by drubinstein** [comment]
Can you add a test dataset loader. Should that be a separate or combined with three test indexes?

**Pull request merged**